src/app/component/CartItem/CartItem.component.js

```diff
@@ -88,9 +96,7 @@ class CartItem extends Component {
     handleRemoveItem() {
         const { removeProduct, product } = this.props;
         this.setState({ isLoading: true });
-        removeProduct({ product }).then(
-            () => this.setState({ isLoading: false })
-        );
+        removeProduct({ product });
```

Comment:
Not inifinite loader, but we cannot set it to false in `.then` clause because after item deletion, component does not exist, so the only thing it causes is memory-leaking.

Reply:
But what if the removal call fails (for whatever reason)? If the item is still there then could do it in `.catch(...)`/`.then(null, ...)`.

Reply:
Thanks for mentioning, for now it seems that it can happen only due to some unhandled error (i.e. that causes error in request). In that case error message from outside component should be shown. Restoring dangerous item from loading state seem too risky in that scenario. However in positive scenario (successful item removal) garbage collection won't work properly as references to callbacks will be from component that does not already exists.